x-pack/plugins/rule_registry/server/search_strategy/search_strategy.ts

```diff
@@ -36,21 +36,37 @@ export const ruleRegistrySearchStrategyProvider = (
   security?: SecurityPluginSetup,
   spaces?: SpacesPluginStart
 ): ISearchStrategy<RuleRegistrySearchRequest, RuleRegistrySearchResponse> => {
-  const es = data.search.getSearchStrategy(ENHANCED_ES_SEARCH_STRATEGY);
-
+  const internalUserEs = data.search.searchAsInternalUser;
+  const requestUserEs = data.search.getSearchStrategy(ENHANCED_ES_SEARCH_STRATEGY);
   return {
     search: (request, options, deps) => {
+      // SIEM uses RBAC fields in their alerts but also utilizes ES DLS which
+      // is different than every other solution so we need to special case
+      // those requests.
+      let siemRequest = false;
+      if (request.featureIds.length === 1 && request.featureIds[0] === AlertConsumers.SIEM) {
+        siemRequest = true;
+      } else if (request.featureIds.includes(AlertConsumers.SIEM)) {
+        throw new Error(
+          'The ruleRegistryAlertsSearchStrategy search strategy is unable to accommodate requests containing multiple feature IDs and one of those IDs is SIEM.'
+        );
+      }
+
       const securityAuditLogger = security?.audit.asScoped(deps.request);
       const getActiveSpace = async () => spaces?.spacesService.getActiveSpace(deps.request);
       const getAsync = async () => {
         const [space, authorization] = await Promise.all([
           getActiveSpace(),
           alerting.getAlertingAuthorizationWithRequest(deps.request),
         ]);
-        const authzFilter = (await getAuthzFilter(
-          authorization,
-          ReadOperations.Find
-        )) as estypes.QueryDslQueryContainer;
+        let authzFilter;
+
+        if (!siemRequest) {
+          authzFilter = (await getAuthzFilter(
+            authorization,
+            ReadOperations.Find
+          )) as estypes.QueryDslQueryContainer;
+        }
```

Comment:
nitpick: might be clearer to return early?
```
if (siemRequest) {
  return { space };
}
...
return {space, authzFilter };
```

Reply:
Hmm. I actually prefer it this way as it's a single place we'd need to update if we wanted to return more things, and as the function gets more complicated later, it might be harder to see the early return